### PR TITLE
Add section on Circle CI permission problems

### PR DESCRIPTION
**Pull request by almarklein**
Also mention the error message verbatim to make this page appear when people search for the problem.
That last bullet is what seemed to make it work for me.

### CODE REVIEW

File: src/ciservices.rst [hunk 1]
```diff
@@ -22,3 +22,16 @@ Circle can be manually added for each fork. Circle calls this "Adding a Project"
...
 This is not normally needed.
+
+Another issue is that Circle sometimes does not have permission to checkout the source code. You'll see a messsage like this::
+
+    Cloning into '.'...
+    Warning: Permanently added the RSA host key for IP address '192.30.253.113' to the list of known hosts.
+    Permission denied (publickey).
+    fatal: Could not read from remote repository.
+
+There does not seem to be a reproducable way to fix this, but here's a list of things that you can try:
```
[the remark this answers is not in the document]
Comment (almarklein) [reply src/ciservices.rst]: @ocefpaf If I understand things correctly, the permission error can be fixed easily with ``@conda-forge-admin, please update circle`` for a feedstock, but for PR's we need manual fiddling as I tried to describe in the bullet points?

File: src/ciservices.rst [hunk 2]
```diff
@@ -22,3 +22,16 @@ Circle can be manually added for each fork. Circle calls this "Adding a Project"
 This effectively amount to going to the `Add Projects <https://circleci.com/add-projects>`_
 page, finding the fork that you wish to enable, and clicking the "Build Project" button.
 This is not normally needed.
+
+Another issue is that Circle sometimes does not have permission to checkout the source code. You'll see a messsage like this::
```
Comment (jakirkham) [inline src/ciservices.rst]: Maybe it is worth saying something like...

> If CircleCI lacks permissions to checkout the source code *in a PR* (as noted in the message below), here are some things you can try...

...or similar.

That should help clarify this is useful for user submitted PRs and not so much for builds on `master` for example.

File: src/ciservices.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (jakirkham) [reply src/ciservices.rst]: Added a suggestion below (or above in file view) for the text, which should help clarify this confusion both for us and users.

**Comment by jakirkham** [comment]
Thanks @almarklein.

**Pull request merged**